src/test/java/edu/cornell/kfs/sys/service/impl/AwsSecretServiceImplIntegraionTest.java

```diff
@@ -0,0 +1,125 @@
+    private static final int BASIC_POJO_NUMBER_VALUE = 1;
+    
+    
+    private AwsSecretServiceImpl awsSecretServiceImpl;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        awsSecretServiceImpl = new AwsSecretServiceImpl();
+        awsSecretServiceImpl.setAwsRegion(AWS_US_EAST_ONE_REGION);
+        awsSecretServiceImpl.setKfsSharedNamespace(KFS_SHARED_NAMESPACE);
+        awsSecretServiceImpl.setKfsInstanceNamespace(KFS_INSTANCE_NAMESPACE);
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        awsSecretServiceImpl = null;
+    }
+
+    @Test
+    void testGetSingleStringValueFromAwsSecret() {
+        String actualSecretValue = awsSecretServiceImpl.getSingleStringValueFromAwsSecret(SINGLE_STRING_SECRET_KEY_NAME, false);
+        assertEquals(SINGLE_STRING_SECRET_VALUE, actualSecretValue);
+    }
+    
+    @Test
+    void testDateSetAndGet() throws ParseException {
+        Date date = new Date(Calendar.getInstance().getTimeInMillis());
+        awsSecretServiceImpl.updateSecretDate(SINGLE_DATE_SECRET_KEY_NAME, false, date);
+        
+        Date secretDate = awsSecretServiceImpl.getSingleDateValueFromAwsSecret(SINGLE_DATE_SECRET_KEY_NAME, false);
+        assertEquals(date.toString(), secretDate.toString());
+    }
+    
+    @Test
+    void testBooleanSetAndGet() {
+        boolean initialValue = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        boolean expectedNewBoolean = !initialValue;
+        awsSecretServiceImpl.updateSecretBoolean(SINGLE_BOOLEAN_SECRET_KEY_NAME, false, expectedNewBoolean);
+        
+        boolean actualNewBoolean = awsSecretServiceImpl.getSingleBooleanFromAwsSecret(SINGLE_BOOLEAN_SECRET_KEY_NAME, false);
+        
+        assertEquals(expectedNewBoolean, actualNewBoolean);
+    }
+    
+    @Test
+    void testPojo() throws JsonMappingException, JsonProcessingException {
+        String newUniqueString = UUID.randomUUID().toString();
+        Date newDate = new Date(Calendar.getInstance().getTimeInMillis());
+        
+        AwsSecretPojo pojo = awsSecretServiceImpl.getPojoFromAwsSecret(BASIC_POJO_SECRET_KEY_NAME, false, AwsSecretPojo.class);
+        LOG.info("testPojo, pojo: " + pojo);
+        pojo.setChangeable_string(newUniqueString);
+        pojo.setUpdate_date(newDate);
+        boolean newBooleanTest = !pojo.isBoolean_test();
+        pojo.setBoolean_test(newBooleanTest);
+        awsSecretServiceImpl.updatePojo(BASIC_POJO_SECRET_KEY_NAME, false, pojo);
+        
+        AwsSecretPojo pojoNew = awsSecretServiceImpl.getPojoFromAwsSecret(BASIC_POJO_SECRET_KEY_NAME, false, AwsSecretPojo.class);
+        assertEquals(newUniqueString, pojoNew.getChangeable_string());
+        assertEquals(BASIC_POJO_STATIC_STRING_VALUE, pojoNew.getStatic_string());
+        assertEquals(BASIC_POJO_NUMBER_VALUE, pojoNew.getNumber_test());
+        assertEquals(newDate.toString(), pojoNew.getUpdate_date().toString());
```

Comment:
Another spot where it would be good to use the date objects themselves in the comparison.

Reply:
switched